hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java

```diff
@@ -393,8 +396,11 @@ private boolean isContainerHealthy(final ContainerInfo container,
    */
   private boolean isContainerUnderReplicated(final ContainerInfo container,
       final Set<ContainerReplica> replicas) {
+    boolean misReplicated = !getPlacementStatus(
```

Comment:
I understand that this is here to check if we have less than 2 racks with replicas, and if so, consider the container to be under replicated.
I am unsure though if we want to ensure that a container is replicated to exactly 2 racks, it does not seems to be the case in the rack aware placement policy, but if so, then mis-replication should be checked somewhere else also to handle the case when a container is replicated to 3 racks due to for example removing a rack from config and place the nodes in a rack to other racks. (Probably this can go to isContainerOverReplicated(), if we want to check for this case.)

Reply:
My thinking on this, is that the policy states the replica "must be on AT LEAST two racks". I don't believe we should consider it over-replicated if it turns out to be on 3 somehow. That logic is consistent with HDFS, and probably provides better redundancy and locality (more racks is likely better in almost all cases).

Reply:
Sounds reasonable for me, let's leave it this way, I wanted to be sure we considered this when it came to my minds as a possible starting state.